web/src/components/task/task-sidebar/finish-button/finish-button.tsx

```diff
@@ -76,45 +89,70 @@ export const FinishButton: FC<TaskSidebarProps> = ({
     );
 
     return !isValidation && viewMode ? null : (
-        <ControlGroup cx={styles['button-finish-control-group']}>
-            <Tooltip content={tooltipContent}>
-                <Button
-                    isDisabled={isDisabled}
-                    cx={styles['button-finish']}
-                    caption={isValidation ? 'FINISH VALIDATION' : 'FINISH LABELING'}
-                    onClick={
-                        isValidation
-                            ? () => {
-                                  handleFinishValidation();
-                              }
-                            : onAnnotationTaskFinish
-                    }
-                />
-            </Tooltip>
-            {isNextTaskPresented && (
-                <PickerInput
-                    valueType="id"
-                    minBodyWidth={322}
-                    selectionMode="single"
-                    value={redirectionSettings}
-                    dataSource={redirectSettingsDataSource}
-                    getName={(item) => item?.caption ?? ''}
-                    onValueChange={handleSetRedirectionSettings}
-                    renderToggler={({ toggleDropdownOpening, ...props }) => (
-                        <Button
-                            {...props}
-                            size="36"
-                            fill="solid"
-                            isDropdown={false}
-                            icon={settingsIcon}
-                            onClear={undefined}
-                            placeholder={undefined}
-                            isDisabled={isDisabled}
-                            onClick={toggleDropdownOpening}
-                        />
-                    )}
-                />
-            )}
-        </ControlGroup>
+        <>
+            <ControlGroup cx={styles['button-finish-control-group']}>
+                <Tooltip content={tooltipContent}>
+                    <Button
+                        isDisabled={isDisabled}
+                        cx={styles['button-finish']}
+                        caption={isValidation ? 'FINISH VALIDATION' : 'FINISH LABELING'}
+                        onClick={() => {
+                            if (confirmWindowNeed === 'true') {
+                                uuiModals
```

Comment:
I propose to move this logic to the separate method to read this code more comfortable (+ we can organise it via async/await in this case - it'll look more compact).

Reply:
fixed